### PR TITLE
Add dev mode kotsadm

### PR DESCRIPTION
**Pull request by syntastical**
#### What this PR does / why we need it:
Added dev mode to kotsadm Okteto

#### Special notes for your reviewer:
NONE

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Does this PR require documentation?
NONE

### CODE REVIEW

File: hack/dev/Dockerfile.okteto [hunk 1]
```diff
@@ -4,19 +4,8 @@ EXPOSE 40000
 
 ENV PROJECTPATH=/go/src/github.com/replicatedhq/kots
```
Comment (marccampbell) [inline hack/dev/Dockerfile.okteto:5]: I think this needs buildkit cache / go mod download pattern

**Review by marccampbell — CHANGES_REQUESTED** [review]

File: hack/dev/Dockerfile.okteto [hunk 2]
```diff
@@ -169,4 +157,6 @@ RUN cd /tmp && curl -fsSL -o helm.tar.gz "${HELM3_URL}" \
 #  && mv s3cmd-${S3CMD_VERSION}/S3 /usr/local/bin/S3 \
 #  && rm -rf s3cmd.tar.gz s3cmd.tar.gz.asc s3cmd-${S3CMD_VERSION}
 
+RUN make kotsadm
```
Comment (pavansokkenagaraj) [inline hack/dev/Dockerfile.okteto:160]: Can we change the make commands to 
```
make build/run/debug
```
to keep it a standard over the projects?

Comment (syntastical) [reply hack/dev/Dockerfile.okteto]: Sadly kots and kotsadm both share this make file so it creates a bit of a naming issue.   I want to get these standardized but I didn't want to go to deep on this PR right now

File: Makefile [hunk 3]
```diff
@@ -56,6 +56,18 @@ mock:
 kotsadm:
 	go build ${LDFLAGS} ${GCFLAGS} -o bin/kotsadm $(BUILDFLAGS) ./cmd/kotsadm
 
+# Debugging
+.PHONY: kotsadm-debug-build
+kotsadm-debug-build:
+	go build ${LDFLAGS} -gcflags="all=-N -l" \
+ 		$(BUILDFLAGS) \
+		-o ./bin/kotsadm-api-debug \
```
Comment (pavansokkenagaraj) [inline Makefile:64]: `bash: BUILDFLAGS: command not found`

File: Makefile [hunk 4]
```diff
@@ -56,6 +56,18 @@ mock:
...
+		./cmd/kotsadm
+
+.PHONY: kotsadm-debug
+kotsadm-debug: kotsadm-debug-build
+	dlv --listen=:2345 --headless=true --api-version=2 exec -- ./bin/kotsadm-api-debug
```
Comment (pavansokkenagaraj) [inline Makefile:69]: https://github.com/replicatedhq/kots/blob/749cceb797f076694b52a31570430540f5c3ca3a/hack/dev/entrypoint.sh#L4
Still building, but I think the dlv command needs `api` as args in end to start the `api` server

**Review by sgalsaleh — APPROVED** [review]

File: hack/dev/Dockerfile.okteto [hunk 5]
```diff
@@ -156,17 +138,17 @@ RUN cd /tmp && curl -fsSL -o helm.tar.gz "${HELM3_URL}" \
...
-#  && mv s3cmd-${S3CMD_VERSION}/s3cmd /usr/local/bin/s3cmd \
-#  && mv s3cmd-${S3CMD_VERSION}/S3 /usr/local/bin/S3 \
-#  && rm -rf s3cmd.tar.gz s3cmd.tar.gz.asc s3cmd-${S3CMD_VERSION}
+RUN go install github.com/go-delve/delve/cmd/dlv@v1.8.0
+
+ENV PROJECTPATH=/go/src/github.com/replicatedhq/kots
+WORKDIR $PROJECTPATH
+
+COPY go.mod go.sum ./
+RUN --mount=target=/go/pkg/mod,type=cache go mod download
```
Comment (marccampbell) [inline hack/dev/Dockerfile.okteto:147]: @syntastical we are going back and forth in repos with the target path here. I'd like to figure out the definite right solution here. Let's chat with Okteto team to get their recommendation? 

I've seen some projects moving to this path, and some moving away from it. I think we need to all be on the same page and in agreement about how we want gomod cached.

**Review by marccampbell — CHANGES_REQUESTED** [review]

**Pull request merged**